javascript/game.js

```diff
@@ -110,11 +159,28 @@ exports.Game = function () {
     );
   };
 
-  var askQuestion = function () {
-    if (currentCategory() == "Pop") console.log(popQuestions.shift());
-    if (currentCategory() == "Science") console.log(scienceQuestions.shift());
-    if (currentCategory() == "Sports") console.log(sportsQuestions.shift());
-    if (currentCategory() == "Rock") console.log(rockQuestions.shift());
+  this.askQuestion = function (category = currentCategory()) {
+    deckArray.forEach((deck) => {
+      if (deck.cat === category) {
+        const position = this.resetIndex(deck.questions, deck.index);
+        deck.index = position;
+        console.log(deck.questions[deck.index]);
+        deck.index += 1; // can use either bracket notation cat['index'] or dot notation cat.index
+      }
+    });
+  };
+
+  //can remove this now
+  this.askQuestionFromCategory = function (questions, index) {
+    console.log(questions[index]);
+  };
+
+  this.resetIndex = function (arr, index) {
```

Comment:
The name of this function is at odds with what the functions does.
The name suggests that the index is reset only whilst the function moves the index along to the next position.